File: 0078-subsets/0078-subsets.java
class Solution {
    private List<List<Integer>> result;

    private void get(int idx, List<Integer> sub, int[] nums) {
        result.add(new ArrayList<>(sub));
        for (int i = idx; i < nums.length; i++) {
            sub.add(nums[i]);
            get(i + 1, sub, nums);
            sub.remove(sub.size() - 1);
        }
    }

    public List<List<Integer>> subsets(int[] nums) {
        result = new ArrayList<>();
        get(0, new ArrayList<>(), nums);
        return result;
    }
}

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        def get(idx, sub):
            result.append(copy.deepcopy(sub))
            for i in range(idx, len(nums)):
                sub.append(nums[i])
                get(i + 1, sub)
                sub.pop()


        result = list()
        get(0, list())
        return result